File: exercises/Exercise.0-Preamble/f90/mpi_hello_world.f90
program contiguous
    use mpi
    implicit none

    integer :: ierr, rank, numtasks, name_len
    character(len=MPI_MAX_PROCESSOR_NAME) :: processor_name
    
    call MPI_INIT(ierr)
    call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
    call MPI_COMM_SIZE(MPI_COMM_WORLD, numtasks, ierr)
    call MPI_Get_processor_name(processor_name, name_len, ierr)

    print *, 'Hello world from rank', rank, 'out of', numtasks, 'on processor', trim(processor_name)

    call MPI_FINALIZE(ierr)

end program

File: exercises/Exercise.0-Preamble/c/mpi_hello_world.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char *argv[])  {
    int numtasks, rank;
    char processor_name[MPI_MAX_PROCESSOR_NAME];
    int name_len;

    MPI_Init(&argc,&argv);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    MPI_Comm_size(MPI_COMM_WORLD, &numtasks);
    MPI_Get_processor_name(processor_name, &name_len);
    
    printf("Hello world from rank %d out of %d on processor %s\n",
           rank, numtasks, processor_name);

    MPI_Finalize();
    return 0;
}
